src/NuGet.Core/NuGet.Packaging/Signing/SignManifest.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using NuGet.Versioning;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Represents the XML manifest containing a set of files and hashes that can be signed.
+    /// </summary>
+    public class SignManifest
+    {
+        public static readonly SemanticVersion DefaultVersion = new SemanticVersion(1, 0, 0);
+
+        public SemanticVersion Version { get; }
+
+        public string GetHash()
```

Comment:
Further, do we need this function? The `singer` can just hash the manifest from the byte stream.